index.js

```diff
@@ -1,31 +1,28 @@
-const String = new Proxy({}, {
-  get: (_, name) => name
-})
+/**
+ * @template Value
+ * @param {(name: string) => Value} mapper
+ * @returns {{ [name: string]: Value }}
```

Comment:
More unsolicited feedback:

This is the first time I see the `@template` tag. I checked and it's not mentioned at [JSDoc.app](https://jsdoc.app/).

But then I found the [`@template` tag in TypeScript's docs](https://www.typescriptlang.org/docs/handbook/jsdoc-supported-types.html#template): it's for declaring generic functions, classes and types.

I would thus recommend renaming `Value` to `T` or `TValue` to clarify that `Values` is generic. It's a common convention to use `T` as the generic parameter name or to prefix generic parameter names with `T`.

---

```suggestion
 * @template T
 * @param {(name: string) => T} mapper
 * @returns {{ [name: string]: T }}
```

or

```suggestion
 * @template TValue
 * @param {(name: string) => TValue} mapper
 * @returns {{ [name: string]: TValue }}
```

(I would personally choose `T` in this case since I don't think that the "Value" in `TValue` brings any extra clarity.)

Reply:
In general single letter names are bad practice even for generics. I could point to examples even within the compiler. In this specific case I agree `Value` doesn't add enough context to merit being different. Accepting option 1 👌🏼

Reply:
> In general single letter names are bad practice even for generics.

I agree, at least if there are more than one generic parameter. `T`, `U`, `R` etc. are confusing. `TFoo`, `TBar`, `TResult` etc. are better.

> I could point to examples even within the compiler.

I would be interested if you have time! Especially if you have examples with only one generic parameter where a single-letter name is bad.

> Accepting option 1 👌🏼

Good choice. 😉